### PR TITLE
fix Qt issue QVariant::value using enum

### PR DESCRIPTION
**Pull request by 3nids**
see https://bugreports.qt.io/browse/QTBUG-53384
fix #19868

### CODE REVIEW

**Comment by nyalldawson** [comment]
What platforms still have qt 5.6? I wonder if we could bump our minimum for 3.6 to something like 5.9...

**Comment by 3nids** [comment]
Ubuntu 16.04 has qt 5.5...

**Comment by nyalldawson** [comment]
Can we drop that for 3.6? They'd still get the 3.4 LTR.

**Comment by 3nids** [comment]
Sounds reasonable to me but that's not my decision to make

**Comment by nyalldawson** [comment]
I'll bring it up on the list

**Comment by nyalldawson** [comment]
I also see this in :
QgsMapLayerStyleCategoriesModel::setData
qgsoptions.cpp:1600
QgsRelationAddDlg::relationStrength()
... + other places too. They should get the same fix.

**Comment by haubourg** [comment]
Hi, Ubuntu 16.04 is on maintenance update cycle up to 2021 . Raising the minimum QT version in 3.6, and fixing 16.04 to QGIS 3.4 makes sense to me. a great +1

File: src/app/qgsvectorlayerloadstyledialog.cpp [hunk 1]
```diff
@@ -111,7 +111,13 @@ QgsMapLayer::StyleCategories QgsVectorLayerLoadStyleDialog::styleCategories() co
 
 QgsVectorLayerProperties::StyleType QgsVectorLayerLoadStyleDialog::currentStyleType() const
 {
+#if QT_VERSION <= 0x050601
+  // since in Qt 5.6.1, QVariant does not correctly convert enum using value
```
Comment (nirvn) [inline src/app/qgsvectorlayerloadstyledialog.cpp]: Is it since 5.6.1 or up to 5.6.1?

Comment (3nids) [reply src/app/qgsvectorlayerloadstyledialog.cpp]: up to 5.6.1
I'll rephrase the comment

**Pull request merged**